sockets/proxy.go

```diff
@@ -1,51 +0,0 @@
-package sockets
-
-import (
-	"net"
-	"net/url"
-	"os"
-	"strings"
-
-	"golang.org/x/net/proxy"
-)
-
-// GetProxyEnv allows access to the uppercase and the lowercase forms of
-// proxy-related variables.  See the Go specification for details on these
-// variables. https://golang.org/pkg/net/http/
-func GetProxyEnv(key string) string {
```

Comment:
I think these are still in use (at least, a quick search showed that `GetProxyEnv` is used in moby/moby https://github.com/moby/moby/blob/9c83848fc95a200c0c69a34cdeeb1f18a090c13a/daemon/info.go#L65-L67

the function itself doesn't look very interesting, but perhaps we need to Deprecate it first

@AkihiroSuda wdyt?

Reply:
+1 on @thaJeztah 's comment

>  which dropped support for golang.org/x/net/proxy. 

Could you be more specific?

Reply:
More specifics are in the commit message, although I guess I should have copied them to the PR comment...

Prior to Go 1.9, golang.org/x/net/proxy was used to add SOCKS support to http.Transport.  Since Go 1.9, SOCKS support has been built-in to the standard http.Transport, so golang.org/x/net/proxy is no longer needed.  In Go 1.11 (https://go-review.googlesource.com/c/go/+/41031/), the golang.org/x/net/proxy library was removed entirely because it had some issues that couldn't be fixed without breaking API compatibility (if users have to touch their code to accommodate these fixes, they might as well just stop using golang.org/x/net/proxy entirely).